pkg/cli/admin/prune/imageprune/prune_test.go

```diff
@@ -95,6 +95,50 @@ func TestImagePruning(t *testing.T) {
 		expectedBlobDeletionFailures         []string
 		expectedErrorString                  string
 	}{
+		{
+			name:             "should not prune old tag revision, still in use",
+			keepTagRevisions: keepTagRevisions(1),
+			images: Images(
+				imagetest.Image(
+					"sha256:0000000000000000000000000000000000000000000000000000000000000000",
+					registryHost+"/foo/bar@sha256:0000000000000000000000000000000000000000000000000000000000000000",
+				),
+				imagetest.Image(
+					"sha256:0000000000000000000000000000000000000000000000000000000000000001",
+					registryHost+"/foo/bar@sha256:0000000000000000000000000000000000000000000000000000000000000001",
+				),
+			),
+			streams: Streams(
+				imagetest.Stream(
+					registryHost,
+					"namespace",
+					"name",
```

Comment:
The test is not correct, the pod uses `foo/bar@sha256:000...01`, but you want to keep `namespace/name@sha256:000...01` because of this.

Reply:
But there is no image `namespace/name@sha256:000...01`, what do you mean? Before this PR the image used by the pod was being deleted.

Reply:
You have only one imagestream: `namespace/name`. And only one pod that uses `foo/bar@...`. The pod uses an imagestream that doesn't exist, so its reference is not valid and can be ignored by the pruner.